cli/data.go

```diff
@@ -0,0 +1,195 @@
+package cli
+
+import (
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	datapb "go.viam.com/api/app/data/v1"
+	rdkutils "go.viam.com/rdk/utils"
+	"google.golang.org/protobuf/encoding/protojson"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+)
+
+const (
+	dataDir     = "data"
+	metadataDir = "metadata"
+)
+
+// BinaryData writes the requested data to the passed directory.
+func (c *AppClient) BinaryData(dst string, filter *datapb.Filter) error {
```

Comment:
[super nit] I think this is short for destination so I would just call it dest. Otherwise, disregard

Reply:
`dst` is aI think the most common abbreviation for destination in Go (and I think other C-based languages though less sure). One example would be [io.Copy](https://pkg.go.dev/io#Copy)